graspologic/cluster/autogmm.py

```diff
@@ -687,8 +714,19 @@ def _process_paramgrid(paramgrid):
             ag_params = {key: params[key] for key in ag_keys}
             if ag_params not in ag_params_processed:
                 ag_params_processed.append(ag_params)
+            if (
+                gm_params not in gm_params_processed
+                and ag_params["affinity"] == "none"
+                and run_multiple_init
+            ):
+                gm_params_processed.append(gm_params.copy())
+                gm_params_processed[-1].update({"n_init": n_init})
```

Comment:
I could be wrong but it seems to me that what's updated here and hence the output `param_grid` on line 538 has {"n_init": n_init} which is not necessarily 1 but the integer defined by the user (which is `self.n_init`)? So it's increasing the number of init for GM instead of adding more runs with `n_init=1`?

Reply:
Basically, here I am storing all the gm_params that will be used for k-means in a list called gm_params_processed. This way I  can keep track of how many k-means run I need to add with multiple init and immeditaly add the number `n_init` to this dictionary. However, these new rows are not added until line 726 to 729 outside of the loop when I append these new rows to paramgrid_processed. Due to the fact that I am using gm_params.copy() at this stage I am not modifying any of the run parameters that would normally be stored. So I am effectively appending later new rows rather than increasing the number of init for GM. I will provide a Notebook to show this. The linked Notebook, should also show that rows are effectively appended.

Reply:
Sorry for the late reply. I see that you are appending the rows. But in each of them, it's `{"n_init": n_init}` instead of 1, right? I tried to run your code and look at the intermediary outputs. They look like what's in Out[183] in [this notebook](https://nbviewer.jupyter.org/github/PerifanosPrometheus/GiorgioDiSalvo/blob/master/autogmm_verification.ipynb) linked in your comment above. Am I looking at the right one?

Reply:
@tliu68 Don't worry about the late reply, I appreciate your time and it is totally understandable! You are correct, I am appending rows with `{"n_init": n_init}`, so that the user has the ability to specify how many additional k-means `n_init` he wishes to perform rather than just one. We can check with @bdpedigo if this is what is intended or if instead I should only append run of k-means with `{"n_init": 1}`. You are looking at the correct Notebook, if from the outputs it isn't particularly clear I would be more than happy to create another one where it may be more clear to see what is happening!

Reply:
i think we want `n_init` rows added for each set of params, and for each, it gets `n_init=1` (but there are `n_init` of them, so the effect is the same). Reason being this lets you see the variance in the initializations. Does that makes sense?

Reply:
Oh I understand, so basically, instead having for each parameter one row with `n_init =3` we want three rows for that same parameter with `n_init=1` so that the model found should be the same but can see how the scores varies per initialization?

Reply:
yes exactly. also i realized this actually matters when using AIC instead of BIC

Reply:
Perfect! Thank you for letting me know : )